### ISSUE TITLE
better error handling for 0-size vector creation

### ISSUE THREAD

**Issue by slayoo** [fynv/ThrustRTC#23]
as of now, passing a zero value for vector size causes the following behaviour:

```python
import ThrustRTC as trtc
trtc.device_vector('double', 0)
cuMemAlloc() failed with Error code: 1
Error Name: CUDA_ERROR_INVALID_VALUE
Error Description: invalid argument
Out[4]: <ThrustRTC.DVVector.DVVector at 0x7f8bca0b1d90>
```

**Comment by fynv** [comment]
Fixed in 0.3.17 release, current behavior:
```
>>> import ThrustRTC as trtc
>>> trtc.device_vector('double', 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\workspace\ThrustRTC_local\install\test_python\ThrustRTC\DVVector.py", line 62, in device_vector
    raise ValueError("0-size vector not allowed")
ValueError: 0-size vector not allowed
```
Also merged pull request #22

**Comment by slayoo** [comment]
Thank you, @fynv !
